### PR TITLE
add optional syn::Error handling

### LINKED ISSUE #3: Provide `syn::Error integration`
Commonly the error message improves, but if used in the case of an `syn::Error` converted into `TokenStream`, the `compile_error!` macro is not pretty. In those cases, and only if the user opted into it, it'd be preferable to not write it to file but error directly without the intermediate file. The integrated span would then point to the source of the error.

### PR DESCRIPTION
**Pull request by drahnr**
Closes #3

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -152,13 +164,20 @@ fn expand_to_file(
...
 
-    let mut shortened = hash.to_hex();
-    shortened.truncate(12);
+    // take the leading 12 hex characters
+    let mut shortened_hex = String::with_capacity(12);
+    const TABLE: &[u8] = b"0123456789abcdef";
+    for &byte in digest.iter().take(6) {
+        shortened_hex.push(TABLE[((byte >> 4) & 0xf) as usize] as char);
+        shortened_hex.push(TABLE[((byte >> 0) & 0xf) as usize] as char);
+    }
```
Comment (sandreim) [inline src/lib.rs]: Move into helper function would be nice.

File: src/lib.rs [hunk 2]
```diff
@@ -101,22 +101,34 @@ impl Expander {
         self
     }
 
+    #[cfg(feature = "syndicate")]
```
Comment (sandreim) [inline src/lib.rs]:
```suggestion
    #[cfg(all(feature = "syndicate", test))]
```

File: Cargo.toml [hunk 3]
```diff
@@ -12,7 +12,13 @@ repository = "https://github.com/drahnr/expander.git"
...
-blake3 = "1.3"
+blake2 = "0.10"
+syn = { version = "1", optional = true, default-features = false }
 
 [dev-dependencies]
 baz = { path = "./tests/baz" }
+syn = { version = "1", features = ["extra-traits", "full"] }
+
+[features]
+default = ["syndicate"]
```
Comment (sandreim) [inline Cargo.toml:23]: This demands some README changes, right ?

**Review by sandreim — APPROVED** [review]
LGTM with a few nits

File: src/lib.rs (hunk 2 above)
Comment (drahnr) [reply src/lib.rs]: Should be `any` rathe than `all`?

File: Cargo.toml (hunk 3 above)
Comment (drahnr) [reply Cargo.toml:23]: Good point!

**Pull request merged**